src/Main.java

```diff
@@ -1,278 +1 @@
-                nextCell.hCost = Heuristic(nextCell, goal);
-                nextCell.gCost = cur.gCost + 1 + nextCell.hCost;
-                if (isValidMoveAStar(maze, nextCell, m, n) && !isCellinList(settledCells, nextCell)) {
-                    if (!isCellinList(unsettledCells, nextCell)) {
-                        unsettledCells.add(nextCell);
-                    } else {
-                        Cell vistedCell = GetCellfromList(unsettledCells, nextCell);
-                        if (nextCell.gCost < vistedCell.gCost){
-                            vistedCell.gCost = nextCell.gCost;
-                            vistedCell.parent = nextCell.parent;
-                        }
-                    }
-
-                }
-            }
-        }
-
-        return null;
-    }
-
-    // Implementation of Manhattan distance
-    static int Heuristic (Cell cell, Cell goal){
-        return Math.abs(goal.x - cell.x) + Math.abs(goal.y - cell.y);
-    }
-
-    static int GetCellwithLowestCost(List<Cell> path){
-        int cost = Integer.MAX_VALUE;
-        int posn = 0;
-        for (int i = 0; i < path.size(); i++){
-            if (path.get(i).gCost < cost) {
-                cost = path.get(i).gCost;
-                posn = i;
-            }
-        }
-        return posn;
-    }
-
-    static Cell GetCellfromList(List<Cell> path, Cell cell){
-        for (int i = 0; i < path.size(); i++){
-            if (CompareCells(path.get(i), cell)) return path.get(i);
-        }
-        return null;
-    }
-
-    static Boolean isCellinList(List<Cell> path, Cell cell){
-        for (int i = 0; i < path.size(); i++){
-            if (CompareCells(path.get(i), cell)) return true;
-        }
-        return false;
-    }
-
-    static Boolean CompareCells(Cell a, Cell b){
-        if (a.x == b.x && a.y == b.y) return true;
-        else return false;
-    }
-
-    static Boolean isValidMove(char[][] maze, Cell cell, int m, int n){
-        // Boundary conditions
-        if (cell.x < 0 || cell.x >= m || cell.y < 0 || cell.y >= n || maze[cell.x][cell.y] == '#'
-                || maze[cell.x][cell.y] == 'X') return false;
-        return true;
-    }
-
-    static Boolean isValidMoveAStar(char[][] maze, Cell cell, int m, int n){
-        // Boundary conditions
-        if (cell.x < 0 || cell.x >= m || cell.y < 0 || cell.y >= n || maze[cell.x][cell.y] == '#') return false;
-        return true;
-    }
-}
-
-public class Main {
-    public static void main(String[] args) {
-        // write your code here
-
-        try {
-            // Specify input path here
-            Scanner sc = new Scanner(new File("C:\\Users\\jimfa\\Desktop\\mazer\\mazer\\test.txt"));
-
-            // Comment this out if you do not want to output to file
-            PrintStream output = new PrintStream(new File("Solved_Maze.txt"));
-            System.setOut(output);
-
-            // Control which kind of algorithm you want to use, please turn only ONE on
```

Comment:
I understand why you are turning only one on, because you mutate the underlying maze when you run through it. But you don't need to do it that way. You should consider copying the maze out several times and having multiple solutions which all work correctly.